api/src/test/java/org/openmrs/module/queue/api/QueueEntryServiceTest.java

```diff
@@ -109,6 +109,48 @@ public void shouldCreateNewQueueEntryRecord() {
 		assertThat(result.getPriority(), is(conceptPriority));
 	}
 	
+	@Test
+	public void shouldNotCreateDuplicateOverlappingQueueEntryRecords() {
+		Queue queue = new Queue();
+		Patient patient = new Patient();
+		Concept conceptStatus = new Concept();
+		Concept conceptPriority = new Concept();
+		Date queueStartDate = new Date();
+		
+		QueueEntry savedQueueEntry = new QueueEntry();
+		savedQueueEntry.setQueueEntryId(QUEUE_ENTRY_ID);
+		savedQueueEntry.setQueue(queue);
+		savedQueueEntry.setPatient(patient);
+		savedQueueEntry.setStatus(conceptStatus);
+		savedQueueEntry.setPriority(conceptPriority);
+		savedQueueEntry.setStartedAt(queueStartDate);
+		
+		QueueEntry duplicateQueueEntry = new QueueEntry();
+		duplicateQueueEntry.setQueue(queue);
+		duplicateQueueEntry.setPatient(patient);
+		duplicateQueueEntry.setStartedAt(queueStartDate);
+		
+		QueueEntrySearchCriteria searchCriteria = new QueueEntrySearchCriteria();
+		searchCriteria.setPatient(patient);
+		searchCriteria.setQueues(Collections.singletonList(queue));
+		
+		when(dao.createOrUpdate(savedQueueEntry)).thenReturn(savedQueueEntry);
+		when(dao.getQueueEntries(searchCriteria)).thenReturn(Collections.singletonList(savedQueueEntry));
+		
+		// Should be able to save and re-save a queue entry without causing validation failure
+		savedQueueEntry = queueEntryService.saveQueueEntry(savedQueueEntry);
+		queueEntryService.saveQueueEntry(savedQueueEntry);
```

Comment:
@cioan - note, I added this test to ensure that saving the same queue entry multiple times doesn't fail validation